Library/Homebrew/formula.rb

```diff
@@ -980,7 +998,7 @@ def migration_needed?
   end
 
   # @private
-  def outdated_versions
+  def outdated_versions(options = {})
     @outdated_versions ||= begin
```

Comment:
I'm not sure but I think that we don't need to fetch remote repository twice when we already have outdated versions with `:fetch_head => true` cached somewhere.

As I can see we only use `Formula#outdated` and `Formula#outdated_versions` in `cmd/upgrade` and `cmd/outdated`. For `outdated` we don't need to reassign anything (at least now). For `upgrade` we  need `outdated` only to understand that it's time to upgrade.

Also, supposing we don't cache anything at all, is it possible that `outdated_versions` returns something different from what it returns on its first call? It seems that no, otherwise we would have the same caching problem we have now. And assuming that our further behaviour depends only on what is returned by `outdated_versions` on its first call, if `outdated_versions(:fetch_head => true)` is not empty on its first call it won't change from what it is on first call. If `outdated_versions(:fetch_head => true)` is empty on its first call, I think it's even worse if it gets not empty somewhere in the middle of the process. So, I think having `outdated_versions` with and without the flag isn't that bad.